editor/editor_node.cpp

```diff
@@ -2692,9 +2692,7 @@ void EditorNode::_edit_current(bool p_skip_foreign, bool p_skip_inspector_update
 				if (!changing_scene) {
 					main_plugin->edit(current_obj);
 				}
-			}
-
-			else if (main_plugin != editor_plugin_screen && (!ScriptEditor::get_singleton() || !ScriptEditor::get_singleton()->is_visible_in_tree() || ScriptEditor::get_singleton()->can_take_away_focus())) {
+			} else if (main_plugin != editor_plugin_screen && editor_main_screen->can_auto_switch_screens()) {
```

Comment:
```suggestion
			} else if (main_plugin != editor_plugin_screen) {
```
This is the wrong place for this change. When trying to open a script resource in the inspector and the current tab is set to Game or Assetlib, the main screen doesn't switch to Script, but it should.
Also, the `text_editor/behavior/navigation/stay_in_script_editor_on_node_selected` editor setting no longer works when set to false. 
This second half of this condition used to always return true, so it should just be the first half here.

Try setting skip_main_plugin above, next to where it is currently being set.

Reply:
Good catch! I moved this next to the `stay_in_script_editor_on_node_selected` code. Specifically, it checks if the script editor is focused, then uses `stay_in_script_editor_on_node_selected`, else uses the general function.